src/libs/actions/Report.js

```diff
@@ -500,9 +500,14 @@ function fetchChatReports() {
  * Get the actions of a report
  *
  * @param {Number} reportID
+ * @param {Number} offset
+ * @returns {Promise}
  */
-function fetchActions(reportID) {
-    API.Report_GetHistory({reportID})
+function fetchActions(reportID, offset) {
+    const reportActionsOffset = _.isNumber(offset)
+        ? offset
+        : Math.max(reportMaxSequenceNumbers[reportID] - CONST.REPORT.REPORT_ACTIONS_LIMIT, -1);
```

Comment:
This should be 0 instead of -1. We should pass `-1` if there's no reportMaxSequenceNumber or however we check to see if we already have a report's chats or not.
Like if you create a new chat with 0 reportActions, reportMaxSequenceNumber should be undefined right? That's when we want to pass -1

Reply:
I'm not sure where that's done though

Reply:
> It seems there are two cases we need to handle:
> 
> Case 1:
> 
> We are explicitly paginating from an existing set of results and in this case should:
> Pass the oldest sequenceNumber we have to fetchActions when we are paginating so we can get the previous set of results. This would be that sequenceNumber minus the default limit (or 0 whichever is larger like @madmax330 is saying) to get the next page.
> 
> Case 2:
> 
> We are not explicitly paginating at all and need to either:
> A. Fetch all the history that we have missed based on the highest sequenceNumber we have stored
> B. Fetch the history with offset -1 and default limit to get limit "most recent" actions

I believe this is what I'm currently doing. Case 1 is done when we call `fetchActions` in `loadMoreChats`:
```JS
fetchActions(this.props.reportID, Math.max(minSequenceNumber - CONST.REPORT.REPORT_ACTIONS_LIMIT, 0))
```

Case 2 is done in the line above:
> A. Fetch all the history that we have missed based on the highest sequenceNumber we have stored
```JS
reportMaxSequenceNumbers[reportID] - CONST.REPORT.REPORT_ACTIONS_LIMIT
```
> B. Fetch the history with offset -1 and default limit to get limit "most recent" actions
```JS
-1
```

Lemme know if I can make this more clear!

Reply:
But yes this problem still exists:

> I think what we are doing now is basically never getting the most recent items for the report since we're taking the most recent thing we have then asking for the previous 50. Using `reportMaxSequenceNumber` like this doesn't make sense since if the highest `sequenceNumber` we have is `100` then we should be able to assume that we already have `limit` actions from that number so in this case `50-100` and we are going to request again what we already have stored and nothing after that?

Reply:
Okay, correct me if I'm wrong but aren't the most recent items of the report being merged in with the reportActions object as they come in?

https://github.com/Expensify/Expensify.cash/blob/a167097f89657b4e0adaec55fff388f79114e406/src/libs/actions/Report.js#L245-L315

Reply:
I think the problem would only exist when closing the app, receiving messages, and reopening it after, like so:
![Kapture 2021-03-01 at 22 58 07](https://user-images.githubusercontent.com/4741899/109611079-8e16f980-7ae2-11eb-8b7a-08c2543892c1.gif)

In which case, we can do the following:
> always fetch the most recent limit actions *no matter what* and then replace all of the actions on the report with the most recent ones.

Only when reopening the app? Thoughts?

Reply:
>correct me if I'm wrong but aren't the most recent items of the report being merged in with the reportActions object as they come in

Yep, that is the "best case scenario". _Hopefully_ those comments show up. In real world scenarios, it probably is not safe to assume that just because the app is open/active/whatever that you are completely up to date on a report's history. You could have a crappy internet connection or Pusher doesn't send an update or something else unexpected could happen. So we rely on the "reconnection callbacks" to bring us up to speed on what we've missed. e.g. this guy here:

https://github.com/Expensify/Expensify.cash/blob/e8084f94737f6a2561bf3cd454b412d93359cb93/src/libs/actions/Report.js#L729-L732

>Only when reopening the app

Ok, so we should replace all the history with the last n items, but _only_ when reopening the app? Why do this instead of just return the most recent items up to the limit?

Reply:
Well, as you said when we reconnect we'll be fetching all the most recent chats anyway. So, the only scenario where we need to refresh the chat and load the new messages is if we're opening it back up again, as in the screengrab I posted above, right?

Reply:
Okay, discussed with @marcaaron - the chats already scroll to the most recent when you toggle between them anyway. So, as he mentioned before let's just fetch the most recent actions whenever you open/switch between reports.

Obviously, this is not ideal, since you lose a chat whenever you toggle out of it and always reload it at its most recent position. But since we already drop you in the most recent position when you toggle between chats anyway, we reason this is okay. Let's make a follow-up issue to explore how to best get away from this pattern though.

Reply:
> let's just fetch the most recent actions whenever you open/switch between reports.

Looks like we're already doing this here: https://github.com/Expensify/Expensify.cash/blob/f6ca82ca208a7d627b62f08054d72c05f6b3e6f3/src/pages/home/report/ReportActionsView.js#L89

I think we just need to make sure we load the actions when we open the app then.

Reply:
`Math.max(reportMaxSequenceNumbers[reportID] - CONST.REPORT.REPORT_ACTIONS_LIMIT, -1);`

This logic seems wrong. If we are trying to get the most recent actions we will need to always pass `-1` for the offset?

Reply:
Largest sequence number we have minus the actions limit will just give us 50 things we _already_ have and nothing new?

Reply:
... Yes. I feel dumb 😛. This only needs to be `-1`.